tests/web.rs

```diff
@@ -164,6 +166,118 @@ mod test {
+        let write_path_array: Array = js_array(&["zero.bin"]).into();
+        let ls_path_array: Array = js_array(&[]).into();
+        let zero_content_buffer = Uint8Array::new_with_length(10);
+        let zero_content_array_buffer = zero_content_buffer.buffer();
+        mount
+            .write(write_path_array.clone(), zero_content_array_buffer.clone())
+            .await?;
+        let ls: Array = mount.ls(ls_path_array.clone()).await?;
+        assert_eq!(ls.length(), 1);
+        let ls_0 = ls.get(0);
+        let fs_entry = WasmFsMetadataEntry::try_from(ls_0).unwrap();
+        assert_eq!(fs_entry.name(), "zero.bin");
+        assert_eq!(fs_entry.entry_type(), "file");
+        let shared_string = mount.share_file(write_path_array).await?;
+
+        let cs = mount.content_blockstore();
+        let new_bytes = client
+            .read_shared_file_from_bs(shared_string, &cs)
+            .await?
+            .to_vec();
+        // Assert successful reconstruction
+        assert_eq!(new_bytes, zero_content_buffer.to_vec());
+
+        Ok(())
+    }
+
+    // // TODO: this is failing due to the server not receiving appropriate updates in a timely manner
+    // #[wasm_bindgen_test]
+    // async fn mount_write_share_receive() -> TombResult<()> {
+    //     let mut client = authenticated_client().await?;
+    //     info!("tomb_wasm_test: create_bucket_mount_write_share_receive()");
+    //     let (private_pem, public_pem) = ecencryption_key_pair().await;
+    //     let bucket_mount =
+    //         create_bucket_and_mount(&mut client, private_pem.clone(), public_pem).await?;
+    //     let mut mount = bucket_mount.mount();
+    //     assert!(!mount.locked());
+    //     let write_path_array: Array = js_array(&["zero.bin"]).into();
+    //     let ls_path_array: Array = js_array(&[]).into();
+    //     let zero_content_buffer = Uint8Array::new_with_length(10);
+    //     let zero_content_array_buffer = zero_content_buffer.buffer();
+    //     mount
+    //         .write(write_path_array.clone(), zero_content_array_buffer.clone())
+    //         .await?;
+    //     let ls: Array = mount.ls(ls_path_array.clone()).await?;
+    //     assert_eq!(ls.length(), 1);
+    //     let ls_0 = ls.get(0);
+    //     let fs_entry = WasmFsMetadataEntry::try_from(ls_0).unwrap();
+    //     assert_eq!(fs_entry.name(), "zero.bin");
+    //     assert_eq!(fs_entry.entry_type(), "file");
+    //     let shared_string = mount.share_file(write_path_array).await?;
+
+    //     // Wait 5 seconds for the write to complete server-side
+    //     gloo_timers::future::TimeoutFuture::new(5_000).await;
+
+    //     let new_bytes = client
+    //         .read_shared_file(shared_string)
+    //         .await
+    //         .expect("read_shared_file_from_bs failed")
+    //         .to_vec();
+
+    //     // Assert successful reconstruction
+    //     assert_eq!(new_bytes, zero_content_buffer.to_vec());
+
+    //     Ok(())
+    // }
```

Comment:
Talked to Alex about these commented out tests, it seems they're likely failing due to the outstanding metadata bug and should be re-enabled when that has been resolved.